src/App.tsx

```diff
@@ -1,61 +1,126 @@
+import React, { useEffect, useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { Todo } from './Types/InterfaceTodo';
+
+import { TodoList } from './components/TodoList';
+
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const preparedTodos = todosFromServer.map(todo => ({
+  ...todo,
+  user: usersFromServer.find(user => user.id === todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedUserID, setSelectedUserID] = useState('0');
+  const [title, setTitle] = useState('');
+  const [isTitleError, setIsTitleError] = useState(false);
+  const [isUserError, setIsUserError] = useState(false);
+
+  useEffect(() => {
+    setTodos(preparedTodos);
+  }, []);
+
+  const addedNewTodo = (newTodo: Todo) => {
+    setTodos([...todos, newTodo]);
+  };
+
+  const onTitleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(e.target.value);
+    setIsTitleError(false);
+  };
+
+  const onUserChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUserID(e.target.value);
+    setIsUserError(false);
+  };
+
+  const onReset = () => {
+    setTitle('');
+    setSelectedUserID('0');
+  };
+
+  const onSubmitForm = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (!title) {
+      setIsTitleError(true);
+    }
+
+    if (selectedUserID === '0') {
+      setIsUserError(true);
+    }
+
+    if (!title || selectedUserID === '0') {
+      return;
+    }
+
```

Comment:
The calculation of `newId` using `Math.max(...preparedTodos.map(todo => todo.id))` assumes that `preparedTodos` is never empty. Consider handling the case where `preparedTodos` might be empty to avoid potential errors.